### DISCUSSION TITLE
Air native extensions for Linux and packagedDependencies

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#1227] [Q&A] [answered]
Hello,

how to add a dependency in a Air native extensions for Linux ?

I tried :

```
adt -package -target ane ... -platform Linux-x86-64 -platformoptions platform.xml libMain.so libTest.so
```

```
<platform xmlns="http://ns.adobe.com/air/extension/33.1">
	<packagedDependencies>
	   <packagedDependency>libTest.so</packagedDependency>
	</packagedDependencies>
</platform>
```

but `-platformoptions platform.xml` is not supported on Linux

Thanks

**Comment by ajwfrost** [comment]
I would have said just include them in a "libs" subfolder of the platform folder there, per the final section of the below documentation:
https://help.adobe.com/en_US/air/extensions/WSf268776665d7970d-2482335412ffea65006-8000.html
but actually it looks like you can just have them in the same folder.

So e.g. create a folder "Linux" into which you put the library.swf, libMain.so, and libTest.so, and then just do:
```
adt -package -target ane ... -platform Linux-x86-64 -C Linux .
```
then it will pick up all the files in that folder and embed them into the ANE.

**Comment by pol2095** [comment, marked as answer]
I needed to add :

```
g++ -shared -fPIC -o EdgeDetect.so EdgeDetect.o -lc -L. -lFlashRuntimeExtensions -lTest -L. -Wl,-rpath,"\$ORIGIN"
```

Thanks
